src/mediapipe_internal/mediapipegraphexecutor.cpp

```diff
@@ -890,4 +890,131 @@ Status MediapipeGraphExecutor::infer(const KFSRequest* request, KFSResponse* res
+    return StatusCode::OK;
+}
+
+Status MediapipeGraphExecutor::inferStream(const ::inference::ModelInferRequest& firstRequest, ::grpc::ServerReaderWriterInterface<::inference::ModelStreamInferResponse, ::inference::ModelInferRequest>& stream) {
+    try {
+        // Init
+        ::mediapipe::CalculatorGraph graph;
+        MP_RETURN_ON_FAIL(graph.Initialize(this->config), "graph initialization", StatusCode::MEDIAPIPE_GRAPH_INITIALIZATION_ERROR);
+
+        // Installing observers
+        for (const auto& name : this->outputNames) {
+            MP_RETURN_ON_FAIL(graph.ObserveOutputStream(name, [&stream, &name](const ::mediapipe::Packet& packet) -> absl::Status {
+                try {
+                    ::inference::ModelStreamInferResponse resp;
+                    // TODO: Add support for other types CVS-122327/CVS-122329
+                    OVMS_RETURN_MP_ERROR_ON_FAIL(receiveAndSerializePacket<ov::Tensor>(packet, *resp.mutable_infer_response(), name), "ov::Tensor serialization");  // TODO: Missing test
+                    resp.mutable_infer_response()->set_id(std::to_string(packet.Timestamp().Value()));
+                    if (!stream.Write(resp)) {
+                        return absl::Status(absl::StatusCode::kCancelled, "client disconnected");
+                    }
+                    return absl::OkStatus();
+                } catch (...) {
+                    return absl::Status(absl::StatusCode::kCancelled, "error in serialization");
+                }
+            }),
+                "output stream observer installation", StatusCode::INTERNAL_ERROR);  // Should never happen for validated graphs
+        }
+
+        // Launch
+        MP_RETURN_ON_FAIL(graph.StartRun({}), "graph start", StatusCode::MEDIAPIPE_GRAPH_START_ERROR);  // TODO: Input side packets / missing test
+
+        // Deserialize first request
+        OVMS_RETURN_ON_FAIL(this->partialDeserialize(
+                                std::shared_ptr<const ::inference::ModelInferRequest>(&firstRequest,
+                                    // Custom deleter to avoid deallocation by custom holder
+                                    // Conversion to shared_ptr is required for unified deserialization method
+                                    // for first and subsequent requests
+                                    [](const ::inference::ModelInferRequest*) {}),
+                                graph),
+            "partial deserialization of first request");
+
+        // Read loop
+        // Here we create ModelInferRequest with shared ownership,
+        // and move it down to custom packet holder to ensure
+        // lifetime is extended to lifetime of deserialized Packets.
+        auto req = std::make_shared<::inference::ModelInferRequest>();
+        while (stream.Read(req.get())) {
+            OVMS_RETURN_ON_FAIL(this->partialDeserialize(req, graph), "partial deserialization of subsequent requests");
+            if (graph.HasError()) {
```

Comment:
If graph has an error, we retrieve its value in WaitUntilDone. It is then logged in debug level via MP_RETURN_ON_FAIL

Reply:
> Above we return on deserialization error. Why do we break in this segment instead of returning as well?

We do break not due to deserialization error, but due to graph error. We need to be able to stop the stream in case the graph encountered an error internally (due to errors in calculators).
We do that by checking graph state between each Read() operations.

Reply:
Added non breaking error reporting for failed deserialization errors